lib/web/apiserver_test.go

```diff
@@ -2002,6 +2002,63 @@ func TestGetMFADevices(t *testing.T) {
 	require.Equal(t, "otp", devices[0].Name) // default device name
 }
 
+func TestGetAndDeleteMFADevices_WithRecoveryApprovedToken(t *testing.T) {
+	t.Parallel()
+	ctx := context.Background()
+	env := newWebPack(t, 1)
+	proxy := env.proxies[0]
+
+	// Create a user with a totp device.
+	username := "llama"
+	proxy.createUser(ctx, t, username, "root", "password", "some-otp-secret")
+
+	// Enable second factor.
+	ap, err := types.NewAuthPreference(types.AuthPreferenceSpecV2{
+		Type:         constants.Local,
+		SecondFactor: constants.SecondFactorOptional,
+		U2F: &types.U2F{
+			AppID:  "https://" + env.server.ClusterName(),
+			Facets: []string{"https://" + env.server.ClusterName()},
+		},
+	})
+	require.NoError(t, err)
+	err = env.server.Auth().SetAuthPreference(ctx, ap)
+	require.NoError(t, err)
+
+	// Acquire an approved token.
+	approvedToken, err := types.NewUserToken("some-token-id")
+	require.NoError(t, err)
+	approvedToken.SetUser(username)
+	approvedToken.SetSubKind(auth.UserTokenTypeRecoveryApproved)
+	approvedToken.SetExpiry(env.clock.Now().Add(5 * time.Minute))
+	_, err = env.server.Auth().Identity.CreateUserToken(context.Background(), approvedToken)
+	require.NoError(t, err)
+
+	// Call the getter endpoint.
+	clt := proxy.newClient(t)
+	getterEndpoint := clt.Endpoint("webapi", "mfa", approvedToken.GetName())
```

Comment:
nit: inline below?

or:

```suggestion
	getDevicesEndpoint := clt.Endpoint("webapi", "mfa", approvedToken.GetName())
```